web/src/thumbnail.ts

```diff
@@ -0,0 +1,39 @@
+import {realm} from "./state_data";
+
+type ThumbnailFormat = {
+    name: string;
+    max_width: number;
+    max_height: number;
+    format: string;
+    animated: boolean;
+};
```

Comment:
You can use `z.infer` to get this type from `zod` without needing to repeat the list that you have in `state_data.ts`.